app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -13,26 +13,23 @@ class AuthorizationController < ApplicationController
     before_action :store_request, only: [:index]
     before_action :override_csp_with_uris, only: [:index]
     before_action :confirm_user_is_authenticated_with_fresh_mfa, only: :index
+    before_action :confirm_user_has_aal3_mfa_if_requested, only: [:index]
     before_action :prompt_for_password_if_ial2_request_and_pii_locked, only: [:index]
     before_action :bump_auth_count, only: [:index]
 
     def index
       return redirect_to_account_or_verify_profile_url if profile_or_identity_needs_verification?
       return redirect_to(sign_up_completed_url) if needs_sp_attribute_verification?
       link_identity_to_service_provider
+      return redirect_to(aal3_required_url) if aal3_required?
       return redirect_to(user_authorization_confirmation_url) if auth_count == 1
-      aal3_requirement || handle_successful_handoff
+      handle_successful_handoff
     end
 
     private
 
-    def aal3_requirement
-      return unless aal3_policy.aal3_required?
-      redirect_to aal3_required_url unless aal3_policy.aal3_used?
-    end
-
-    def aal3_policy
-      @aal3 ||= AAL3Policy.new(session)
+    def aal3_required?
```

Comment:
This one confused me a bit. Thoughts on changing the name to `aal3_required_but_not_used?` or something like that?